### PR TITLE
p-dml: implement rollback for pipelined dml

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: ref #50215

Problem Summary:

Implement rollback for pipelined dml.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

#### Flush phase

image

#### Rollback phase

image

#### SQL client

image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by cfzjywxk — APPROVED** [review]

**Review by zyguan — APPROVED** [review]

**Review by crazycs520 — APPROVED** [review]

File: pkg/metrics/grafana/tidb.json [hunk 1]
```diff
@@ -19023,7 +19023,7 @@
           "steppedLine": false,
           "targets": [
             {
-              "expr": "histogram_quantile(0.95, sum(rate(tidb_tikvclient_range_task_push_duration_bucket{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\"}[1m])) by (le, instance))",
+              "expr": "histogram_quantile(0.95, sum(rate(tidb_tikvclient_range_task_push_duration_bucket{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\", type=~\"resolve-locks.*\"}[1m])) by (le, instance))",
```
Comment (ekexium) [inline pkg/metrics/grafana/tidb.json:19026]: How about filtering by something like `!~".*pipelined-dml.*"`

Comment (you06) [reply pkg/metrics/grafana/tidb.json]: Done, test it ok

image

**Pull request merged**